.github/workflows/main.yml

```diff
@@ -0,0 +1,559 @@
+
+        for dependency in "${dependency_list[@]}"; do
+          which "${dependency}"
+          "${dependency}" --version
+
+          printf "\n\n"
+        done
+
+        /usr/local/osquery-toolchain/usr/bin/clang --version
+
+    - name: Select the build job count
+      shell: bash
+      id: build_job_count
+      run: |
+        echo ::set-output name=VALUE::$(($(nproc) + 1))
+
+    - name: Select the build options for the tests
+      shell: bash
+      id: tests_build_settings
+      run: |
+        if [[ "${{ matrix.build_type }}" == "RelWithDebInfo" ]] ; then
+          echo ::set-output name=VALUE::OFF
+        else
+          echo ::set-output name=VALUE::ON
+        fi
+
+    - name: Select the debug symbols options
+      shell: bash
+      id: debug_symbols_settings
+      run: |
+        if [[ "${{ matrix.build_type }}" == "Debug" ]] ; then
+          echo ::set-output name=VALUE::OFF
+        else
+          echo ::set-output name=VALUE::ON
+        fi
+
+    # When we spawn in the container, we are root; create an unprivileged
+    # user now so that we can later use it to launch the normal user tests
+    - name: Create a non-root user
+      if: matrix.build_type != 'RelWithDebInfo'
+      id: unprivileged_user
+      run: |
+        useradd -m -s /bin/bash unprivileged_user
+        echo ::set-output name=NAME::unprivileged_user
+
+    # Due to how the RPM packaging tools work, we have to adhere to some
+    # character count requirements in the build path vs source path.
+    #
+    # Failing to do so, will break the debuginfo RPM package.
+    #
+    # Symbolic links were being dereferenced, so the `mount -o bind` approach
+    # was chosen instead. The `path` parameter introduced in the checkout@v2
+    # action could easily solve this problem, but we lose access to the `.git`
+    # folder so it's not an option.
+    - name: Setup the build paths
+      id: build_paths
+      run: |
+        mkdir -p workspace/usr/src/debug/osquery/build \
+                 workspace/ccache \
+                 workspace/padding-required-by-rpm-packages/src
+
+        chown -R ${{ steps.unprivileged_user.outputs.NAME }}:${{ steps.unprivileged_user.outputs.NAME }} .
+
+        mount -o bind "$(pwd)" workspace/padding-required-by-rpm-packages/src
+
+        echo ::set-output name=SOURCE::$(realpath workspace/padding-required-by-rpm-packages/src)
+        echo ::set-output name=BINARY::$(realpath workspace/usr/src/debug/osquery/build)
+        echo ::set-output name=CCACHE::$(realpath workspace/ccache)
+
+    # One of the tests in the test suit will spawn a Docker container
+    # using this socket. Allow the unprivileged user we created
+    # to access it.
+    - name: Update the Docker socket permissions
+      if: matrix.build_type != 'RelWithDebInfo'
+      run: |
+        chmod 666 /var/run/docker.sock
+
+    - name: Update the cache (ccache)
+      uses: actions/cache@v2
```

Comment:
Caveat -- this is documented as having 5gb of cache per repo. Is that enough?

Reply:
Yeah, by default ccache is configured to 5GB too and there's plenty to have a quick build.
Actually, we might have to keep and eye on the available free space because when filled up, you need 2x that space for the task to tar and upload it.

Reply:
👍